File: exercicios em C/q39.cpp
#include <stdio.h>

int main(void){
	int i, n, idade, media, idadeMaior, idadeMenor, quantMaior, quantMenor;
	idadeMaior = 0;
	idadeMenor = 0;
	quantMaior = 0;
	quantMenor = 0;
	printf("\n\nEste programa ler a idade de n pessoas e comparaas para assim definir, a idade media, quem tem a maior idade, quem tem a menor idade, quantos sao maiores de idades, e quantos sao menores de idade.");
	printf("\n\nDigite a quantidade de pessoas a qual iremos comparar a idade:");
	scanf("%d", &n);
	for (i = 1; i <= n; i++){
		printf("\n\nDigite quantos anos a %d pessoa tem: ", i);
		scanf("%d", &idade);
		media = media + idade;
		if (i == 1){
			idadeMenor = idade;
			idadeMaior = idade;
		}
		if (idadeMaior < idade){
			idadeMaior = idade;
		}
		else if(idadeMenor > idade){
			idadeMenor = idade;
		}
		if (idade > 18){
			quantMaior = quantMaior + 1;
		}
		if(idade <= 18){
			quantMenor = quantMenor + 1;
		}
	}
	media = media / n;
	printf("\n\nA media idade das %d pessoa e igual a %d, a pessoa mais velha tem %d anos de idade, e mais nova tem %d anos de idade.", n, media, idadeMaior, idadeMenor);
	printf("\nSomando todas as pessoas maiores de idade resulta em %d pessoas, e somando todas menores de idade e igual a %d pessoas.", quantMaior, quantMenor);
	printf("\nObrigado por utilizar nosso sistema!");
	return 0;
}

File: exercicios em Python/q39.py
import os

print("\n\nEste programa ler a idade de n pessoas e comparaas para assim definir, a idade media, quem tem a maior idade, quem tem a menor idade, quantos sao maiores de idades, e quantos sao menores de idade.")
n = int(input("\nDigite a quantidade de pessoas a qual iremos comparar a idade:"))
i = 1
media = 0
quantMaior = 0
quantMenor = 0
while i <= n:
        print("\nDigite quantos anos a ",i," pessoa tem: ")
        idade = int(input(""))
        media = media + idade
        if i == 1:
                idadeMenor = idade
                idadeMaior = idade
                
        if idadeMaior < idade:
                idadeMaior = idade
                
        elif idadeMenor > idade:
                idadeMenor = idade

        if idade >= 18:
                quantMaior = quantMaior + 1

        if idade < 18:
                quantMenor = quantMenor + 1

        i = i + 1
        
media = int(media / n)
print("\nA media idade das ",n," pessoa e igual a ",media,", a pessoa mais velha tem ",idadeMaior," anos de idade, e mais nova tem ",idadeMenor," anos de idade.")
print("\nSomando todas as pessoas maiores de idade resulta em ",quantMaior," pessoas, e somando todas menores de idade e igual a ",quantMenor," pessoas.")
print("\nObrigado por utilizar nosso sistema!")
os.system("pause")
